apps/heft/src/utilities/TaskPackageResolver.ts

```diff
@@ -0,0 +1,254 @@
+      rigPackageFolder,
+      terminal
+    );
+    const eslintPackagePath: string | undefined = TaskPackageResolver._tryResolveTaskPackage(
+      'eslint',
+      projectFolder,
+      rigPackageFolder,
+      terminal
+    );
+    const apiExtractorPackagePath: string | undefined = TaskPackageResolver._tryResolveTaskPackage(
+      '@microsoft/api-extractor',
+      projectFolder,
+      rigPackageFolder,
+      terminal
+    );
+
+    return {
+      apiExtractorPackagePath,
+      typeScriptPackagePath,
+      tslintPackagePath,
+      eslintPackagePath
+    };
+  }
+
+  private static _tryResolveTaskPackage(
+    taskPackageName: string,
+    projectFolder: string,
+    rigPackageFolder: string | undefined,
+    terminal: Terminal
+  ): string | undefined {
+    let result: string | undefined = undefined;
+    if (rigPackageFolder) {
+      result = TaskPackageResolver._tryResolvePackage(taskPackageName, rigPackageFolder, terminal, true);
+    }
+    if (!result) {
+      result = TaskPackageResolver._tryResolvePackage(taskPackageName, projectFolder, terminal, false);
+    }
+    return result;
+  }
+
+  private static _tryResolvePackage(
+    taskPackageName: string,
+    baseFolder: string,
+    terminal: Terminal,
+    isRigFolder: boolean
+  ): string | undefined {
+    try {
+      if (isRigFolder) {
+        terminal.writeVerboseLine(`Attempting to resolve "${taskPackageName}" from rig folder ${baseFolder}`);
+      } else {
+        terminal.writeVerboseLine(`Attempting to resolve "${taskPackageName}" from ${baseFolder}`);
+      }
+      const resolvedPackageJsonFile: string = resolve.sync(taskPackageName, {
+        basedir: baseFolder,
+        preserveSymlinks: false,
+        packageFilter: (packageJson: INodePackageJson) => {
+          return {
+            ...packageJson,
+            // ensure "main" points to a file in the package root folder
+            main: 'package.json'
+          };
+        }
+      });
+      const resolvedPackageFolder: string = path.dirname(resolvedPackageJsonFile);
+
+      if (isRigFolder) {
+        terminal.writeVerboseLine(
+          `Resolved "${taskPackageName}" via rig package to ${resolvedPackageFolder}`
+        );
+      } else {
+        terminal.writeVerboseLine(`Resolved "${taskPackageName}" to ${resolvedPackageFolder}`);
+      }
+
+      return resolvedPackageFolder;
+    } catch (e) {
+      // Ignore errors
+    }
+    return undefined;
+  }
+
+  private static _locateRigPackageFolder(
+    tsconfigPath: string,
+    visitedTsconfigPaths: Set<string>,
+    visitedRigPackagePaths: Set<string>,
+    terminal: Terminal
+  ): string | undefined {
+    if (visitedTsconfigPaths.has(tsconfigPath)) {
+      throw new Error(`The file "${tsconfigPath}" has an "extends" field that creates a circular reference`);
+    }
+    visitedTsconfigPaths.add(tsconfigPath);
+
+    terminal.writeVerboseLine(`Examining ${tsconfigPath}`);
+
+    if (!FileSystem.exists(tsconfigPath)) {
```

Comment:
Since JsonFile.load throws when it's not found and the exception can be checked with `FileSystem.isFileNotFoundError`, we can skip this check and do it all in the following try/catch

Reply:
Fixed